pkg/ottl/ottlfuncs/func_parse_csv_test.go

```diff
@@ -0,0 +1,500 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func Test_ParseCSV(t *testing.T) {
+	tests := []struct {
+		name        string
+		oArgs       ottl.Arguments
+		want        map[string]any
+		createError string
+		parseError  string
+	}{
+		/* Test default mode */
+		{
+			name: "Parse comma separated values",
+			oArgs: &ParseCSVArguments[any]{
+				Target: ottl.StandardStringGetter[any]{
+					Getter: func(ctx context.Context, tCtx any) (any, error) {
+						return "val1,val2,val3", nil
+					},
+				},
+				Header: ottl.StandardStringGetter[any]{
+					Getter: func(ctx context.Context, tCtx any) (any, error) {
+						return "col1,col2,col3", nil
+					},
+				},
+			},
+			want: map[string]any{
+				"col1": "val1",
+				"col2": "val2",
+				"col3": "val3",
+			},
+		},
+		{
+			name: "Parse with newline in first field",
+			oArgs: &ParseCSVArguments[any]{
+				Target: ottl.StandardStringGetter[any]{
+					Getter: func(ctx context.Context, tCtx any) (any, error) {
+						return "val1\nnewline,val2,val3", nil
```

Comment:
I think all the tests have the newline somewhere in the middle of the field value. Could you add a test for something like `val1\n,val2,val3` where the newline is at the end? Having it at the beginning like `\nval1,val2,val3` would probably be good too.